File: notif-ingester/models/transaction_vector.py
# Source: Python (Original)
from datetime import datetime, timezone
from pydantic import BaseModel, Field
from uuid_extensions import uuid7
from typing import List

class TransactionVector(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid7()))
    user_id: str = Field(default="default", alias="UserId")
    transaction_id: str
    vendor: str
    category: str
    summary: str = ""
    debit_account_id: str
    credit_account_id: str
    embed_text: str
    embedding: List[float]
    confirmed_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    partition_key: str = "default"

    class Config:
        populate_by_name = True

File: backend/Models/TransactionVector.cs
// Source: p/models/transaction_vector.py
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace FinanceApp.Models
{
    public class TransactionVector
    {
        [JsonPropertyName("id")]
        public string Id { get; set; } = string.Empty;

        [JsonPropertyName("UserId")]
        public string UserId { get; set; } = "default";

        [JsonPropertyName("transaction_id")]
        public string TransactionId { get; set; } = string.Empty;

        [JsonPropertyName("vendor")]
        public string Vendor { get; set; } = string.Empty;

        [JsonPropertyName("category")]
        public string Category { get; set; } = string.Empty;

        [JsonPropertyName("summary")]
        public string Summary { get; set; } = string.Empty;

        [JsonPropertyName("debit_account_id")]
        public string DebitAccountId { get; set; } = string.Empty;

        [JsonPropertyName("credit_account_id")]
        public string CreditAccountId { get; set; } = string.Empty;

        [JsonPropertyName("embed_text")]
        public string EmbedText { get; set; } = string.Empty;

        [JsonPropertyName("embedding")]
        public List<float> Embedding { get; set; } = new();

        [JsonPropertyName("confirmed_at")]
        public DateTime ConfirmedAt { get; set; } = DateTime.UtcNow;

        [JsonPropertyName("partition_key")]
        public string PartitionKey { get; set; } = "default";
    }
}
